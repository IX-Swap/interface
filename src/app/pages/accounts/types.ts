import { Bank, Address } from 'types/bank'
import { Asset } from 'types/asset'
import { PaginationArgs } from 'services/api/types'

export type BankFormValues = Pick<
  Bank,
  'bankName' | 'accountHolderName' | 'bankAccountNumber' | 'swiftCode'
> & {
  asset: Asset['_id']
  address: Bank['address']
}

export type AddressValues = Address

export interface TransactionWithOTP {
  otp: string
}

export interface TransactionBase {
  amount: number
}

export interface DepositCashFormValues
  extends TransactionBase,
    TransactionWithOTP {
  asset: string
}

export interface WithdrawCashFormValues {
  amount?: number | null
  otp?: string | null
  bankAccountId?: string | null
  memo?: string
  paymentMethodName: string | null
}
export interface WithdrawCashParams extends WithdrawCashFormValues {
  virtualAccount: string
}
export interface DepositDSFormValues {
  balanceId: string
}

export interface WithdrawDSFormValues
  extends TransactionBase,
    TransactionWithOTP {
  recipientWallet: string
  memo?: string
}

export type UpdateBankArgs = BankFormValues & {
  bankId: string
}

export interface DepositCashArgs {
  depositCode: string
  asset: string
  amount: number
  otp: string
  memo?: string
}

export type WithdrawCashArgs = WithdrawCashFormValues

export type WithdrawDSArgs = WithdrawDSFormValues & { asset: string }

export interface GetBanksArgs extends PaginationArgs {}

export interface GetWithdrawalAddressesArgs extends PaginationArgs {}
