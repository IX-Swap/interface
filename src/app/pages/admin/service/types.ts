type Base = Record<string, string | undefined>

export interface UpdateUserRoleArgs extends Base {
  userId?: string
  roles: string
}
