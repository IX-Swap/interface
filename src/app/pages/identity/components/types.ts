import {
  PersonalProfileWithAddress,
  IdentityFinancials,
  BaseIdentity,
  CorporateFields
} from 'types/identity'
import { DataroomFile, FormArray } from 'types/dataroomFile'
import { Maybe } from 'types/util'
import {
  CorporateDeclarations,
  IndividualDeclarations
} from 'app/pages/identity/const/declarations'

export type IndividualIdentityFormValues = PersonalProfileWithAddress &
  IdentityFinancials &
  Omit<
    BaseIdentity,
    | '_id'
    | 'status'
    | 'user'
    | 'createdAt'
    | 'updatedAt'
    | 'documents'
    | 'declarations'
  > & {
    documents: FormArray<Maybe<DataroomFile>>
    declarations: IndividualDeclarations
  }

export type CorporateIdentityFormValues = CorporateFields &
  Omit<
    BaseIdentity,
    | '_id'
    | 'status'
    | 'user'
    | 'createdAt'
    | 'updatedAt'
    | 'documents'
    | 'declarations'
  > & {
    documents: FormArray<Maybe<DataroomFile>>
    declarations: CorporateDeclarations
  }
