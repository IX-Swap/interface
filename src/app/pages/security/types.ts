export interface Enable2faFormValues {
  otp: string
}

export interface Remove2faFormValues {
  otp: string
  emailCode: string
}

export interface TwoFaData {
  encoded: string
  image: string
  key: string
}

export interface GetEmailCodeResponse {
  email: string
}
