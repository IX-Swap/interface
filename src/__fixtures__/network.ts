import { Network } from 'types/networks'

export const network: Network = {
  nativeCurrency: {
    name: 'Ropsten Ether',
    symbol: 'ETH',
    decimals: 18
  },
  explorer: {
    urls: {
      home: 'https://ropsten.etherscan.io/',
      transaction: 'https://ropsten.etherscan.io/tx/%s',
      address: 'https://ropsten.etherscan.io/address/%s',
      token: 'https://ropsten.etherscan.io/token/%s'
    },
    name: 'Etherscan for Rospten',
    description:
      'Etherscan is the leading BlockChain Explorer, Search, API and Analytics Platform for Ethereum, a decentralized smart contracts platform.'
  },
  supportsEvm: true,
  hasUTXO: false,
  _id: '5f88035d7ae447ee9274d4fa',
  name: 'Ropsten Test Network (Public Ethereum)',
  chainId: 3,
  createdAt: '2020-10-15T08:07:57.264Z',
  isSandbox: true,
  networkId: 3,
  nodes: [
    {
      _id: '5f88035d6709c90de75b1e9f',
      name: 'Infura Provider for Ropsten',
      description:
        "Infura is a hosted Ethereum node cluster that lets your users run your application without requiring them to set up their own Ethereum node or wallet. You may not be familiar with Infura by name, but if you've used MetaMask then you've used Infura, as it is the Ethereum provider that powers MetaMask.",
      url: 'https://infura.io/',
      endpoint: 'https://ropsten.infura.io/v3/aa62950f7d3942ec9a6b46a7586d0f35',
      params: [
        {
          _id: '5f88035d6709c90de75b1ea0',
          key: null,
          value: null
        }
      ],
      headers: [
        {
          _id: '5f88035d6709c90de75b1ea1',
          key: null,
          value: null
        }
      ],
      user: null,
      password: null,
      meta: {
        secretUserAgent: 'InvestaX Blockchain Services v0.1',
        projectId: 'aa62950f7d3942ec9a6b46a7586d0f35',
        projectSecret: 'c44d27f2d3954bfbb4545a9aa8608746',
        bearer: ''
      }
    }
  ],
  updatedAt: '2020-10-15T08:07:57.264Z'
}

const algorandNetwork: Network = {
  nativeCurrency: {
    name: 'Algorand',
    symbol: 'ALGO',
    decimals: 0
  },
  explorer: {
    urls: {
      home: 'https://goalseeker.purestake.io',
      transaction:
        'https://goalseeker.purestake.io/algorand/testnet/transaction/%s',
      address: 'https://goalseeker.purestake.io/algorand/testnet/account/%s',
      token: 'https://goalseeker.purestake.io/algorand/testnet/search/%s'
    },
    name: 'Goalseeker for algorand',
    description:
      'Goalseeker is the leading BlockChain Explorer, Search, API and Analytics Platform for Algorand, a decentralized smart contracts platform.'
  },
  supportsEvm: true,
  hasUTXO: false,
  _id: '611a2458c8f34bf9cc34589c',
  name: 'ALGOD V2.2.0 TESTNET',
  chainId: 6,
  createdAt: '2021-08-16T08:39:52.372Z',
  isSandbox: true,
  networkCode: 'ALGO_TESTNET',
  networkId: 6,
  nodes: [],
  updatedAt: '2021-08-16T08:39:52.372Z'
}

export const networks: Network[] = [
  network,
  {
    nativeCurrency: {
      name: 'Ether',
      symbol: 'ETH',
      decimals: 18
    },
    explorer: {
      urls: {
        home: 'https://etherscan.io/',
        transaction: 'https://etherscan.io/tx/%s',
        address: 'https://etherscan.io/address/%s',
        token: 'https://etherscan.io/token/%s'
      },
      name: 'Etherscan for Rospten',
      description:
        'Etherscan is the leading BlockChain Explorer, Search, API and Analytics Platform for Ethereum, a decentralized smart contracts platform.'
    },
    supportsEvm: true,
    hasUTXO: false,
    _id: '5f88035d7ae447ee9274d4fb',
    name: 'Main Ethereum Network',
    chainId: 1,
    createdAt: '2020-10-15T08:07:57.264Z',
    isSandbox: true,
    networkId: 1,
    nodes: [
      {
        _id: '5f88035d6709c90de75b1ea2',
        name: 'Infura Provider for Ethereum Mainnet',
        description:
          "Infura is a hosted Ethereum node cluster that lets your users run your application without requiring them to set up their own Ethereum node or wallet. You may not be familiar with Infura by name, but if you've used MetaMask then you've used Infura, as it is the Ethereum provider that powers MetaMask.",
        url: 'https://infura.io/',
        endpoint:
          'https://mainnet.infura.io/v3/aa62950f7d3942ec9a6b46a7586d0f35',
        params: [
          {
            _id: '5f88035d6709c90de75b1ea3',
            key: null,
            value: null
          }
        ],
        headers: [
          {
            _id: '5f88035d6709c90de75b1ea4',
            key: null,
            value: null
          }
        ],
        user: null,
        password: null,
        meta: {
          secretUserAgent: 'InvestaX Blockchain Services v0.1',
          projectId: 'aa62950f7d3942ec9a6b46a7586d0f35',
          projectSecret: 'c44d27f2d3954bfbb4545a9aa8608746',
          bearer: ''
        }
      }
    ],
    updatedAt: '2020-10-15T08:07:57.264Z'
  },
  algorandNetwork
]
