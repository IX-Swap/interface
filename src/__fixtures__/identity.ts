import { user } from './user'
import { address, authorizationInfo } from './authorizer'
import { CorporateIdentity, IndividualIdentity } from 'v2/types/identity'
import declarations from 'v2/app/pages/identity/const/declarations'
import { DataroomFile } from 'v2/types/dataroomFile'

export const corporate: CorporateIdentity = {
  _id: '1',
  logo: '',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  documents: [],
  declarations: [],
  walletAddress: 'address',
  status: 'Submitted',
  beneficialOwners: [],
  companyAddress: address,
  companyLegalName: 'InvestaX',
  countryOfFormation: 'Singapore',
  dateOfIncorporation: '01-01-2000',
  directors: [],
  registrationNumber: '123456',
  representatives: [],
  toArrangeCustody: true,
  email: '',
  contactNumber: '',
  user: '',
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: []
}

export const individual: IndividualIdentity = {
  photo: '',
  _id: '1',
  email: 'email@example.com',
  annualIncome: '100000',
  bankAccountName: '',
  bankAccountNumber: '',
  bankName: '',
  contactNumber: '1234567890',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  countryOfResidence: 'Russian Federation',
  dob: 'DOB',
  employer: 'InvestaX',
  employmentStatus: 'Employed',
  firstName: 'John',
  lastName: 'Doe',
  gender: 'M',
  houseHoldIncome: '100000',
  industryOfEmployment: 'IT',
  maritalStatus: 'Married',
  middleName: '',
  nationality: 'Russian',
  occupation: 'Occupied',
  // politicallyExposed: false,
  sourceOfWealth: '___',
  status: 'Submitted',
  authorization: authorizationInfo,
  authorizationDocuments: [],
  authorizations: [],
  toArrangeCustody: true,
  walletAddress: '1234567890_',
  declarations: declarations.individual.map(({ key }) => ({
    [key]: undefined
  })),
  documents: [],
  address,
  user: ''
}

export const document: DataroomFile = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  originalFileName: 'documentfile1',
  title: 'Document 1',
  type: 'doctype1',
  user: user._id,
  url: 'https://docurl/'
}

export const documents: DataroomFile[] = [
  { ...document, _id: '1' },
  { ...document, _id: '2' }
]
