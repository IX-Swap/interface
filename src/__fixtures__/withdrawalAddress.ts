import { WithdrawalAddress } from 'types/withdrawalAddress'
import { individual, corporate } from './identity'
import { network } from './network'

export const withdrawalAddress: WithdrawalAddress = {
  _id: 'id123',
  status: 'Approved',
  address: '0x67ed490d810c41263758e7355cef720ffed68cbc',
  identity: { individual, corporates: [corporate] },
  label: 'test label',
  network: network,
  memo: 'test memo',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000'
}
