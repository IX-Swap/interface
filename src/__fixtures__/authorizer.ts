import { Bank } from 'types/bank'
import { APIServiceResponse } from 'services/api/types'
import { CashDeposit } from 'types/cashDeposit'
import { Asset } from 'types/asset'
import { CorporateIdentity, IndividualIdentity } from 'types/identity'
import { user } from '__fixtures__/user'
import { CashWithdrawal } from 'types/cashWithdrawal'
import { DSWithdrawal } from 'types/dsWithdrawal'
import { DigitalSecurityOffering, DSOInsight } from 'types/dso'
import { AuthorizationInfo } from 'types/authorizer'
import { Commitment } from 'types/commitment'
import { emptyFile } from '__fixtures__/file'
import { network } from './network'
import { withdrawalAddress } from './withdrawalAddress'

export const asset: Asset = {
  _id: '5f732c538a568b50914d8372',
  deleted: false,
  createdBy: '5f73240373d4ab4b15fc1b2e',
  symbol: 'SGD',
  name: 'Singapore Dollars',
  type: 'Currency',
  numberFormat: {
    currency: 'SGD'
  },
  createdAt: '2020-09-29T12:45:07.411Z',
  updatedAt: '2020-10-05T03:14:17.244Z',
  description: 'test asset description'
}

export const address = {
  city: 'Omsk',
  country: 'Russian Federation',
  state: 'Siberia',
  line1: 'Address line 1',
  line2: 'Address line 2',
  postalCode: '123456'
}

export const authorizationInfo: AuthorizationInfo = {
  authorizer: 'id',
  comment: '',
  sharedWithUser: false,
  timestamp: ''
}

export const corporate: CorporateIdentity = {
  _id: '1',

  logo: '',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  documents: [],
  declarations: [],
  status: 'Submitted',
  beneficialOwners: [],
  companyAddress: address,
  companyLegalName: 'InvestaX',
  countryOfFormation: 'Singapore',
  dateOfIncorporation: '01-01-2000',
  directors: [],
  registrationNumber: '123456',
  representatives: [],
  email: '',
  contactNumber: '',
  user,
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: []
}

export const individual: IndividualIdentity = {
  photo: '',
  _id: '1',
  email: 'email@example.com',
  annualIncome: '100000',
  bankAccountName: '',
  bankAccountNumber: '',
  bankName: '',
  contactNumber: '1234567890',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  countryOfResidence: 'Russian Federation',
  dob: 'DOB',
  employer: 'InvestaX',
  employmentStatus: 'Employed',
  firstName: 'John',
  lastName: 'Doe',
  gender: 'M',
  houseHoldIncome: '100000',
  industryOfEmployment: 'IT',
  maritalStatus: 'Married',
  middleName: '',
  nationality: 'Russian',
  occupation: 'Occupied',
  // politicallyExposed: false,
  sourceOfWealth: '___',
  status: 'Submitted',
  authorization: authorizationInfo,
  authorizationDocuments: [],
  authorizations: [],
  declarations: [],
  documents: [],
  address,
  user
}

export const dsWithdrawal: DSWithdrawal = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  user: 'user',
  amount: 100000,
  hold: 'hold',
  level: '1',
  memo: 'memo',
  status: 'Approved',
  recipientWallet: '0000000',
  asset,
  transaction: '',
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: [],
  identity: {
    individual,
    corporates: []
  }
}

export const dsoInsight: DSOInsight = {
  activityCount: 65,
  approvedcommitmentCount: 1,
  collectedOn: '2020-11-24T17:06:23.555Z',
  commitmentCount: 0,
  commitmentTotal: 0,
  investorCount: 1,
  raisedMax: 400,
  raisedMin: 400,
  raisedTotal: 400
}

export const dso: DigitalSecurityOffering = {
  _id: '1',
  asset: 'asset',
  businessModel: 'business model',
  capitalStructure: 'capital structure',
  network: network._id,
  corporate,
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  createdBy: 'created by',
  currency: asset,
  deleted: false,
  deploymentInfo: undefined,
  distributionFrequency: 'distribution frequency',
  dividendYield: 1,
  documents: [],
  equityMultiple: 0,
  fundraisingMilestone: 'fundraising milestone',
  grossIRR: 0,
  interestRate: 1,
  introduction: 'introduction',
  investmentPeriod: 1,
  investmentStructure: 'investment structure',
  issuerName: 'issuer name',
  launchDate: '01-01-2000',
  leverage: 0,
  logo: '5f4f7d87f3e2c40bbab8a3f1',
  minimumInvestment: 100,
  pricePerUnit: 1,
  subscriptionDocument: emptyFile,
  team: [],
  tokenName: 'token name',
  tokenSymbol: 'token symbol',
  totalFundraisingAmount: 100000,
  useOfProceeds: 'use of proceeds',
  policyBuilder: undefined,
  user: user._id,
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: [],
  identity: {
    individual,
    corporates: []
  },
  status: 'Approved',
  insight: dsoInsight
}

export const commitment: Commitment = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  status: 'Approved',
  currency: asset,
  createdBy: 'user',
  dso,
  identity: {
    individual,
    corporates: [corporate]
  },
  totalAmount: 0,
  authorizationDocuments: [],
  authorizationOverrides: [],
  authorizationOverride: undefined,
  authorization: undefined,
  authorizations: [],
  hold: 'dsadas',
  level: undefined,
  numberOfUnits: 0,
  pricePerUnit: 0,
  signedSubscriptionDocument: emptyFile,
  user,
  withdrawalAddress: withdrawalAddress
}

export const bank: Bank = {
  _id: '1',
  status: 'Approved',
  deleted: false,
  bankName: 'Rocketbank',
  bankAccountNumber: '1234567890',
  accountHolderName: 'Oleg Tinkoff',
  swiftCode: 'SWIFTCODE',
  authorized: true,
  authorizationDocuments: [],
  authorizations: [],
  supportingDocuments: [],
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  authorization: authorizationInfo,
  identity: {
    corporates: [corporate],
    individual
  },
  user,
  address,
  currency: asset
}

export const cashDeposit: CashDeposit = {
  _id: '1',
  amount: 10000,
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  depositCode: 'DEPOSIT',
  status: 'Approved',
  user: '???',
  bankAccount: bank,
  individual,
  corporates: [corporate],
  asset,
  identity: {
    individual,
    corporates: [corporate]
  },
  authorizations: [],
  authorizationDocuments: [],
  authorization: undefined
}

export const cashWithdrawal: CashWithdrawal = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  user: 'user',
  amount: 100000,
  corporates: [],
  hold: 'hold',
  level: '1',
  memo: 'memo',
  status: 'Approved',
  bank,
  individual,
  asset,
  identity: {
    individual,
    corporates: []
  },
  authorizations: [],
  authorizationDocuments: [],
  authorization: undefined
}

export const authorizerURLs = {
  approve: `/${bank._id}/approve`,
  reject: `/${bank._id}/reject`
}

export const approveResponseSuccess: APIServiceResponse = {
  message: 'Approved successfully',
  success: true
}

export const approveResponseFailure: APIServiceResponse = {
  message: 'Could not approve',
  success: false
}

export const rejectResponseSuccess: APIServiceResponse = {
  message: 'Rejected successfully',
  success: true
}

export const rejectResponseFailure: APIServiceResponse = {
  message: 'Could not reject',
  success: false
}
