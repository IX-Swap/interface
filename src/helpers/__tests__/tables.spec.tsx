import {
  getIndividualLastName,
  getCorporateLegalName,
  getCorporateRepresentativeName,
  renderIndividualOrCompanyName,
  renderBalance,
  renderUSDPrice,
  renderSGDPrice,
  renderCurrencyLabel,
  RenderBolderText,
  renderStatus
} from '../tables'
import { individual, corporate } from '__fixtures__/identity'
import { commitment, cashDeposit } from '__fixtures__/authorizer'
import { WithdrawalAddress } from 'types/withdrawalAddress'
import { network } from '__fixtures__/network'
import {
  CorporateIdentity,
  IndividualIdentity
} from 'app/pages/identity/types/forms'
import { render } from 'test-utils'
import { firstTableItems } from 'ui/UIKit/TablesKit/FirstTable/FirstTable'
import React from 'react'

describe('renderIndividualOrCompanyName', () => {
  it('returns individual name for IndividualIdentity', () => {
    expect(renderIndividualOrCompanyName('Jane', individual)).toBe(
      `Jane ${individual.lastName}`
    )
  })

  it('returns representative name for CorporateIdentity', () => {
    expect(renderIndividualOrCompanyName('Jane', corporate)).toBe('InvestaX')
  })

  it('returns company name for CorporateIdentity if firstName is undefined', () => {
    expect(renderIndividualOrCompanyName(undefined, corporate)).toBe(
      `${corporate.companyLegalName}`
    )
  })

  it('returns individual name for Commitment', () => {
    expect(renderIndividualOrCompanyName('Jane', commitment)).toBe(
      `Jane ${commitment.identity.individual.lastName}`
    )
  })

  it('returns individual name for WithdrawalAddress', () => {
    const withdrawalAddress: WithdrawalAddress = {
      _id: 'id123',
      status: 'Approved',
      address: '0x67ed490d810c41263758e7355cef720ffed68cbc',
      identity: { individual, corporates: [corporate] },
      label: 'test label',
      network: network,
      memo: 'test memo',
      createdAt: '01-01-2000',
      updatedAt: '01-01-2000',
      authorizations: []
    }

    expect(renderIndividualOrCompanyName('Jane', withdrawalAddress)).toBe(
      `Jane ${withdrawalAddress.identity?.individual.lastName ?? ''}`
    )
  })

  it('returns company name for Commitment if individual does not exist', () => {
    expect(
      renderIndividualOrCompanyName('Jane', {
        ...commitment,
        identity: {
          individual: undefined as any,
          corporates: commitment.identity.corporates
        }
      })
    ).toBe(`${commitment.identity.corporates[0].companyLegalName}`)
  })

  it('returns individual name for CashDeposit', () => {
    expect(renderIndividualOrCompanyName('Jane', cashDeposit)).toBe(
      `Jane ${cashDeposit.individual.lastName}`
    )
  })

  it('returns company name for CashDeposit if individual does not exist', () => {
    expect(
      renderIndividualOrCompanyName('Jane', {
        ...cashDeposit,
        individual: undefined as any,
        corporates: cashDeposit.corporates
      })
    ).toBe(`${cashDeposit.corporates[0].companyLegalName}`)
  })
})

describe('getIndividualLastName', () => {
  it('returns empty string if individual is undefined', () => {
    expect(
      getIndividualLastName(undefined as unknown as IndividualIdentity)
    ).toBe('')
  })

  it('returns lastName of individual', () => {
    expect(getIndividualLastName(individual)).toBe(individual.lastName)
  })
})

describe('getCorporateLegalName', () => {
  it('returns empty string if corporate is undefined', () => {
    expect(
      getCorporateLegalName(undefined as unknown as CorporateIdentity)
    ).toBe('')
  })

  it('returns legalName of corporate', () => {
    expect(getCorporateLegalName(corporate)).toBe(corporate.companyLegalName)
  })
})

describe('getCorporateRepresentativeName', () => {
  it('returns empty string if corporate is undefined', () => {
    expect(
      getCorporateRepresentativeName(undefined as unknown as CorporateIdentity)
    ).toBe('')
  })

  it("returns first representative's lastName", () => {
    expect(getCorporateRepresentativeName(corporate)).toBe('')
  })
})

describe('renderBalance', () => {
  it('renders balance in correct way', () => {
    const { container } = render(<>{renderBalance('30', firstTableItems[0])}</>)

    expect(container).toMatchSnapshot()
  })
})
describe('renderSGDPrice', () => {
  it('renders SGD price in correct way', () => {
    const { container } = render(<>{renderSGDPrice('30')}</>)

    expect(container).toMatchSnapshot()
  })
})
describe('renderUSDPrice', () => {
  it('renders USD price in correct way', () => {
    const { container } = render(<>{renderUSDPrice('30')}</>)

    expect(container).toMatchSnapshot()
  })
})
describe('renderCurrencyLabel', () => {
  it('renders currency label', () => {
    const { container } = render(<>{renderCurrencyLabel('SGD')}</>)

    expect(container).toMatchSnapshot()
  })
})
describe('RenderBolderText', () => {
  it('renders bolder text', () => {
    const { container } = render(<>{RenderBolderText('test')}</>)

    expect(container).toMatchSnapshot()
  })
})

describe('renderStatus', () => {
  it('renders status with type', () => {
    const { container } = render(<>{renderStatus('Connected')}</>)

    expect(container).toMatchSnapshot()
  })
})
