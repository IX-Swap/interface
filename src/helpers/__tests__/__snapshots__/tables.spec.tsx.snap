// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderBolderText renders bolder text 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1zdpvo-MuiTypography-root"
  >
    test
  </p>
</div>
`;

exports[`renderBalance renders balance in correct way 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1zdpvo-MuiTypography-root"
  >
    30
  </p>
   
  USD
</div>
`;

exports[`renderCurrencyLabel renders currency label 1`] = `
<div>
  <div
    class="MuiBox-root css-70qvj9"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1zdpvo-MuiTypography-root"
    >
      SGD
    </p>
  </div>
</div>
`;

exports[`renderSGDPrice renders SGD price in correct way 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1zdpvo-MuiTypography-root"
  >
    30
  </p>
    SGD
</div>
`;

exports[`renderStatus renders status with type 1`] = `
<div>
  <div
    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorSuccess MuiChip-filledSuccess makeStyles-wrapper-1 makeStyles-wrapper-2 css-bpa8yc-MuiChip-root"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-1dybbl5-MuiChip-label"
    >
      Connected
    </span>
  </div>
</div>
`;

exports[`renderUSDPrice renders USD price in correct way 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-1zdpvo-MuiTypography-root"
  >
    30
  </p>
    USD
</div>
`;
